Spigot-Server-Patches/0418-Ticket-level-per-player-view-distance-API.patch

```diff
@@ -0,0 +1,525 @@
+                                 ChunkMapDistance.this.addTicket(i, ticket);
+                                 ChunkMapDistance.this.l.add(i);
+                             } else {
+@@ -430,10 +529,33 @@ public abstract class ChunkMapDistance {
+                             }
+ 
+                         });
+-                        this.a(i, k, this.c(j), this.c(k));
++                    }
++
++                    //Paper start ticket level pp-view-dist
++                    byte f = ChunkMapDistance.this.update.remove(i);
++                    if(f == 1) {
++                        this.a(i, k, false, true, Integer.MAX_VALUE);
++                    } else if (f == -1) {
++                        this.a(i, k, true, false, Integer.MAX_VALUE);
+                     }
+                 }
+ 
++                ObjectIterator<it.unimi.dsi.fastutil.longs.Long2ByteMap.Entry> it = ChunkMapDistance.this.update.long2ByteEntrySet().iterator();
++
++                while(it.hasNext()) {
++                    it.unimi.dsi.fastutil.longs.Long2ByteMap.Entry e = it.next();
++                    long i = e.getLongKey();
++                    byte f = e.getByteValue();
++                    if(f == 1) {
++                        this.a(i, this.c(i), false, true, Integer.MAX_VALUE);
++                    } else if (f == -1) {
++                        this.a(i, this.c(i), true, false, Integer.MAX_VALUE);
++                    }
++                }
++
++                ChunkMapDistance.this.update.clear();
++                //Paper end
++
+                 this.g.clear();
+             }
+ 
+diff --git a/src/main/java/net/minecraft/server/EntityEnderDragon.java b/src/main/java/net/minecraft/server/EntityEnderDragon.java
+index c8c74f2b3..63f19c510 100644
+--- a/src/main/java/net/minecraft/server/EntityEnderDragon.java
++++ b/src/main/java/net/minecraft/server/EntityEnderDragon.java
+@@ -570,9 +570,9 @@ public class EntityEnderDragon extends EntityInsentient implements IMonster {
+                 // CraftBukkit start - Use relative location for far away sounds
+                 // this.world.b(1028, new BlockPosition(this), 0);
+                 // Paper start
+-                int viewDistance = ((WorldServer) this.world).spigotConfig.viewDistance * 16; // Paper - updated to use worlds actual view distance incase we have to uncomment this due to removal of player view distance API
++                //int viewDistance = ((WorldServer) this.world).spigotConfig.viewDistance * 16; // Paper - updated to use worlds actual view distance incase we have to uncomment this due to removal of player view distance API
+                 for (EntityPlayer player : ((WorldServer)world).getPlayers()) {
+-                    //final int viewDistance = player.getViewDistance(); // TODO apply view distance api patch
++                    final int viewDistance = player.getTrackViewDistance(); //Paper ticket level pp-view-dist
```

Comment:
should be in blocks, check the commented line

Reply:
will be committed thanks